Add instructions to run analyses with npx

yassin-kammoun-sonarsource opened the pull request.

ilia-kebets-sonarsource commented on README.md:
````diff
@@ -27,6 +27,12 @@ To install the scanner globally and be able to run analyses on the command line:
 npm install -g sonarqube-scanner
 ```
 
+To run analyses without installing the scanner at all, run the following command instead:
````
```suggestion
To run analyses without explicitly installing the scanner, run the following command instead:
```

ilia-kebets-sonarsource replied:
I'm slightly uncomfortable having this under `Installation`, would it fit better under `Usage` or somewhere else?

ilia-kebets-sonarsource reviewed (changes requested):
see comments

yassin-kammoun-sonarsource commented:
> I'm slightly uncomfortable having this under Installation, would it fit better under Usage or somewhere else?

Makes sense. What about now?

ilia-kebets-sonarsource commented on README.md:
````diff
@@ -90,6 +84,16 @@ sonar-scanner
+npx sonarqube-scanner
+```
+
+Similar to above, you can specify analysis properties and settings using either a `package.json` file, a `sonar-project.properties` file, or command line arguments.
````
```suggestion
Similar to the above, you can specify analysis properties and settings using either a `package.json` file, a `sonar-project.properties` file, or command line arguments.
```

ilia-kebets-sonarsource reviewed (approved):
check my suggestion, otherwise LGTM!

yassin-kammoun-sonarsource commented:
@ilia-kebets-sonarsource I pushed two additional commits:

- I noticed that the versions of the `package.json` dependencies were not fixed.
- I added TypeScript as a devDependency because TypeScript ESLint seems to need it.

I am not sure why TypeScript was not a devDependency until now. The build was failing because it was missing:

```
added 383 packages in 5.686s

> sonarqube-scanner@3.0.0 check-format /tmp/cirrus-ci-build
> prettier --list-different "{src,test}/**/!(*.lint).js"


> sonarqube-scanner@3.0.0 license /tmp/cirrus-ci-build
> eslint --ext js,ts src test tools scripts .


Oops! Something went wrong! :(

ESLint: 8.30.0

Error: Failed to load parser '@typescript-eslint/parser' declared in '.eslintrc.js': Cannot find module 'typescript'
Require stack:
- /tmp/cirrus-ci-build/node_modules/@typescript-eslint/typescript-estree/dist/convert.js
- /tmp/cirrus-ci-build/node_modules/@typescript-eslint/typescript-estree/dist/ast-converter.js
- /tmp/cirrus-ci-build/node_modules/@typescript-eslint/typescript-estree/dist/parser.js
- /tmp/cirrus-ci-build/node_modules/@typescript-eslint/typescript-estree/dist/index.js
- /tmp/cirrus-ci-build/node_modules/@typescript-eslint/parser/dist/parser.js
- /tmp/cirrus-ci-build/node_modules/@typescript-eslint/parser/dist/index.js
```

How come it didn't fail before? Am I missing something here?

I am requesting another review :)
